src/Pulsar.Client.Otel/OTelProducerInterceptor.fs

```diff
@@ -0,0 +1,61 @@
+﻿module Pulsar.Client.Otel.OTelProducerInterceptor
+
+open System
+open System.Collections.Generic
+open System.Diagnostics
+open OpenTelemetry
+open Pulsar.Client.Api
+open OpenTelemetry.Context.Propagation
+
+//https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Api/README.md#instrumenting-a-libraryapplication-with-net-activity-api
+type OTelProducerInterceptor<'T>() =
+    let Propagator = Propagators.DefaultTextMapPropagator
+    
+    let setter = Action<Dictionary<string,string>,string,string>(fun msg key value -> msg.Add(key,value))
+    static let  source = "pulsar.producer"
+    member this.activitySource : ActivitySource = new ActivitySource(source)
+    static member Source
+        with get() = source
+    interface IProducerInterceptor<'T> with        
+        member this.BeforeSend(producer, message) =
+            let mutableDict = Dictionary<string,string>(message.Properties)
+            let name = producer.Topic + " send"            
+            let activity = this.activitySource.StartActivity(name,ActivityKind.Producer)   
```

Comment:
Is ActivitySource threadsafe? Can we start activities concurrently?

Reply:
I was not able to google anything about it. Classes appeared in .net 5, not much info. I switched from member to "let" for activity source, I hope it's better?